src/sagas/wallet.js

```diff
@@ -456,11 +459,21 @@ export function* handleTx(action) {
   const tx = action.payload;
   const wallet = yield select((state) => state.wallet);
   const routerHistory = yield select((state) => state.routerHistory);
+  const receivedTxs = yield select((state) => state.receivedTxs);
 
   if (!wallet.isReady()) {
     return;
   }
 
+  if (receivedTxs.has(tx.tx_id)) {
+    // This transaction is being processed for a second time.
+    // This may happen due to an event being received twice.
+    // We will skip to avoid multiple notifications and UI changes.
```

Comment:
When do you clean this `receivedTxs` redux state? Because if you receive an update in 2 minutes, it should update this tx, right? It was not a duplicated event.

Reply:
The event will still be treated by the wallet-lib, so we will not ignore it (even with the actual duplicate ones) this will just avoid sendind a "You've received a new transaction" notification which (even with updates) should be avoided.

The other effect would be that we do not request the history update, this request is just to update the redux state history with the history on the wallet-lib.
The only real effect I can think of are voided transactions, but we can add a special case for those and keep the skip on the notification, what do you think°